Cargo.toml

```diff
@@ -21,3 +21,12 @@ travis-ci = { repository = "jonhoo/rust-evmap" }
 maintenance = { status = "passively-maintained" }
 
 [dependencies]
+hashbrown = { version = "0.1.8", optional = true }
+parking_lot = { version = "0.7.1", optional = true }
+smallvec = { version = "0.6.7", optional = true }
+
+[features]
+default = []
```

Comment:
I think we'd probably want to include all the features by default.

Reply:
Perhaps the first two, but not `smallvec`. Of course `evmap` usually encourages small values like pointers, but with `smallvec` a physically large `V` would bloat up the hashmaps even when not used. There is also a tiny logical overhead of using `smallvec`, which is usually worth it if you don’t allocate, but a regular `Vec` would be better if you always do. So while in my project I would enable the feature, others may rather not.

Reply:
Yeah, that's a fair point. I do wonder whether this might lead to surprising results though. For example, imagine I am writing a crate that uses `evmap` and has some large `V`, and therefore intentionally does not enable `smallvec`. Then, someone uses my crate in a larger project that also depends on `evmap` _with_ `smallvec`. Cargo will resolve this situation by enabling the `smallvec` feature, and as a result, performance on my crate will be sad. This is a result of having this be a feature at all, not it being on/off by default though, so maybe that's just something we'll have to accept. But I do think it argues for enabling this by default, and add a warning to disable if you have large `V`, since the common case is likely that `V` is small.

Reply:
It wouldn't be *sad*, just a single extra branch. `smallvec` is most useful for it's memory locality. Combined with `hashbrown`, the entire map is basically a single flat array, which makes the CPU cache very happy.

Reply:
Which I think speaks for `smallvec` + `hashbrown` being the default features. And (as discussed elsewhere), `hashbrown` not even being a feature.